stockflux-securities-master/src/components/InputForm.js

```diff
@@ -15,7 +19,7 @@ const InputForm = ({ match }) => {
   const [enabled, setEnabled] = useState(false);
   const [inProgress, setInProgress] = useState(false);
   const [isLoading, setIsLoading] = useState(false);
-  const [errorMessage, setErrorMessage] = useState(null);
+  const [messages, setMessages] = useState({ type: "success", content: null });
```

Comment:
Maybe it would be better to have two items of state in the form:
- One that describes the current *state*/*context* of the form, or the *type* of messages it has (errors, success etc.) which could be some form of enum type (It's generally not good practice to rely on strings as identifiers, they're very error prone)
- One that stores an array of messages (which could be empty if there are none)

You can then use that state to decide to show messages if there are any and give it the *context* which describes the type of messages they are. You could even use this *state*/*context* item to determine if the form is in the *loading* state, or in the *progress* state. This would help you remove quite a bit of the state and simplify the rendering of the form.

Reply:
That sounds very similar to how I had it before you suggested I change it!

Before I had _success: true or false_ and sendingMessages = [Messages]

I like the idea of re-using the state/context to show loading though. I'll give that a go and see if it works

Reply:
I've changed the state to have one value for loading, sending, error and success. I've created an object to be used as an enum to make it safer than using strings.